### PR TITLE
[MM-32943] commands/user: add promote/demote user commands

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Add promote/demote user commands.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32943

### CODE REVIEW

File: commands/user.go [hunk 1]
```diff
@@ -174,6 +174,24 @@ var VerifyUserEmailWithoutTokenCmd = &cobra.Command{
 	Args:    cobra.MinimumNArgs(1),
 }
 
+var PromoteGuestToUserCmd = &cobra.Command{
+	Use:     "promote [guests]",
+	Short:   "Promote guests to users",
+	Long:    "Convert a guest into a regular user.",
+	Example: "  user promote guest1",
```
Comment (mgdelacroix) [inline commands/user.go:181]: can we use the new example format to show that this can be used with one or multiple users?

File: commands/user.go [hunk 2]
```diff
@@ -174,6 +174,24 @@ var VerifyUserEmailWithoutTokenCmd = &cobra.Command{
...
+	RunE:    withClient(promoteGuestToUserCmdF),
+	Args:    cobra.MinimumNArgs(1),
+}
+
+var DemoteUserToGuestCmd = &cobra.Command{
+	Use:     "demote [users]",
+	Short:   "Demote users to guests",
+	Long:    "Convert a regular user into a guest.",
+	Example: "  user demote user1",
```
Comment (mgdelacroix) [inline commands/user.go:190]: same here

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1: Nonblocking comments related to the commands examples

**Review by lindy65 — APPROVED** [review]
Thanks @isacikgoz - looks good to me 👍 

Please add Zephyr test codes to sRun lines as per below?
image

Thanks! :)

File: go.sum [hunk 3]
```diff
@@ -290,7 +290,6 @@ github.com/golang/protobuf v1.4.1/go.mod h1:U8fpvMrcmy5pZrNK1lt4xCsGvpyWQ/VVv6QD
 github.com/golang/protobuf v1.4.2 h1:+Z5KGCizgyZCbGh1KZqA0fcLLkwbsjIzS4aV2v7wJX0=
 github.com/golang/protobuf v1.4.2/go.mod h1:oDoupMAO8OvCJWAcko0GGGIgR6R6ocIYbsSw735rRwI=
 github.com/golang/snappy v0.0.0-20180518054509-2e65f85255db/go.mod h1:/XxbfmMg8lxefKM7IXC3fBNl/7bRcc72aCRzEWrmP2Q=
```
[the remark this answers is not in the document]
Comment (isacikgoz) [reply go.sum]: Sorted out, sorry for the late answer :) This one escaped from me.

**Pull request merged**